packages/react-charting/src/components/VerticalBarChart/VerticalBarChartRTL.test.tsx

```diff
@@ -0,0 +1,447 @@
+    yAxisCalloutData: '87%',
+    lineData: {
+      y: 16000,
+    },
+  },
+  {
+    x: 92000,
+    y: 45000,
+    legend: 'Monkeys',
+    color: DefaultPalette.blueLight,
+    xAxisCalloutData: '2020/04/30',
+    yAxisCalloutData: '33%',
+    lineData: {
+      y: 40000,
+      yAxisCalloutData: '45%',
+    },
+  },
+];
+
+const simplePoints = [
+  {
+    x: 'This is a medium long label. ',
+    y: 3500,
+    color: '#627CEF',
+  },
+  {
+    x: 'This is a long label This is a long label',
+    y: 2500,
+    color: '#C19C00',
+  },
+  {
+    x: 'This label is as long as the previous one',
+    y: 1900,
+    color: '#E650AF',
+  },
+  {
+    x: 'A short label',
+    y: 2800,
+    color: '#0E7878',
+  },
+];
+
+describe('Vertical bar chart rendering', () => {
+  test('Should render the vertical bar chart with numeric x-axis data', () => {
+    const { container } = render(<VerticalBarChart data={chartPoints} />);
+    expect(container).toMatchSnapshot();
+  });
+
+  test('Should render the vertical bar chart with string x-axis data', () => {
+    const { container } = render(<VerticalBarChart data={simplePoints} />);
+    expect(container).toMatchSnapshot();
+  });
+});
+
+describe('Vertical bar chart - Subcomponent bar', () => {
+  test('Should render the bar with the given width', async () => {
+    // Arrange
+    render(<VerticalBarChart data={chartPoints} barWidth={100} />);
+    await new Promise(resolve => setTimeout(resolve));
+    const bar = screen.getAllByText((content, element) => element!.tagName.toLowerCase() === 'rect')[1];
+
+    // Assert
+    expect(bar.getAttribute('width')).toEqual('100');
+  });
+  test('Should render the bars with the specified colors', async () => {
+    // Arrange
+    // colors mentioned in the data points itself
+    render(<VerticalBarChart data={chartPoints} />);
+    await new Promise(resolve => setTimeout(resolve));
+    const bars = screen.getAllByText((content, element) => element!.tagName.toLowerCase() === 'rect');
+
+    // Assert
+    expect(bars[0].getAttribute('fill')).toEqual('#0078d4');
+    expect(bars[1].getAttribute('fill')).toEqual('#002050');
+    expect(bars[2].getAttribute('fill')).toEqual('#00188f');
+  });
+  test('Should render the bars with the a single color', async () => {
+    // Arrange
+    render(<VerticalBarChart data={chartPoints} useSingleColor={true} />);
+    await new Promise(resolve => setTimeout(resolve));
+    const bars = screen.getAllByText((content, element) => element!.tagName.toLowerCase() === 'rect');
+
+    // Assert
+    expect(bars[0].getAttribute('fill')).toEqual('#00bcf2');
+    expect(bars[1].getAttribute('fill')).toEqual('#00bcf2');
+    expect(bars[2].getAttribute('fill')).toEqual('#00bcf2');
+  });
+  test('Should render the bars with labels hidden', async () => {
+    // Arrange
+    const { container } = render(<VerticalBarChart data={chartPoints} hideLabels={true} />);
+    await new Promise(resolve => setTimeout(resolve));
```

Comment:
Can you tell when promise is needed and when it is not needed.

Reply:
Yes, promise is needed when we are wither trying to provide any prop and update or we are trying to extract any sub-sub svg element like bars within the vertical bar chart.

Reply:
Do we have a high level testing guide doc applicable for all components. This line can go there as a guide.

Reply:
added a high level testing guide.